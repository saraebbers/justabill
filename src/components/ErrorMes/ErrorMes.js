import React from 'react';
import '../../index.scss';


const ErrorMes = (props) => {

  return(
    <div className='ErrorMes-msg'>
      <h3>
        An error has occured, for details, see the message below.  A page refresh may resolve the problem.  
      </h3>
      <h3> 
        {props.message}
      </h3>
    </div>

  )
}

export default ErrorMes;