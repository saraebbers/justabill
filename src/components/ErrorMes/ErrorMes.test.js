import React from 'react';
import ErrorMes from './ErrorMes';
import { shallow } from 'enzyme';

describe('ErrorMes', () => {
  it('ErrorMes should match the snapshot', () => {
    let mockErrorMessage = 'Something went wrong with the Fetch Call'
    const wrapper = shallow(<ErrorMes errorMessage={mockErrorMessage}/>)
    expect(wrapper).toMatchSnapshot()
  })

})
