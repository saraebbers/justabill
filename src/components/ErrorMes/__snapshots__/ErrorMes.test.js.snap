// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMes ErrorMes should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMes
    errorMessage="Something went wrong with the Fetch Call"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          An error has occured, for details, see the message below.  A page refresh may resolve the problem.
        </h3>,
        <h3 />,
      ],
      "className": "ErrorMes-msg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "An error has occured, for details, see the message below.  A page refresh may resolve the problem.",
        },
        "ref": null,
        "rendered": "An error has occured, for details, see the message below.  A page refresh may resolve the problem.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            An error has occured, for details, see the message below.  A page refresh may resolve the problem.
          </h3>,
          <h3 />,
        ],
        "className": "ErrorMes-msg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "An error has occured, for details, see the message below.  A page refresh may resolve the problem.",
          },
          "ref": null,
          "rendered": "An error has occured, for details, see the message below.  A page refresh may resolve the problem.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
